File: bot.py
import discord
from discord.ext import commands
from aiohttp import web
import os
import asyncio

# --- Tiny web server to keep Render container alive ---
async def handle(request):
    return web.Response(text="Bot is running!")
app = web.Application()
app.router.add_get('/', handle)
runner = web.AppRunner(app)

async def start_server():
    await runner.setup()
    site = web.TCPSite(runner, '0.0.0.0', 3000)
    await site.start()
    print("Web server running on port 3000")

# --- Discord bot setup ---
intents = discord.Intents.default()
intents.message_content = True
bot = commands.Bot(command_prefix='.', intents=intents)

leaderboard = {}

@bot.event
async def on_ready():
    print(f'Logged in as {bot.user}!')

@bot.event
async def on_message(message):
    if message.author.bot:
        return

    user_id = message.author.id
    leaderboard[user_id] = leaderboard.get(user_id, 0) + 1

    if message.content == '.lb':
        top = sorted(leaderboard.items(), key=lambda x: x[1], reverse=True)[:10]
        msg = "🏆 Leaderboard 🏆\n" + "\n".join(f"<@{uid}> - {pts} pts" for uid, pts in top)
        await message.channel.send(msg)

    if message.content == '.points':
        await message.channel.send(f"You have {leaderboard[user_id]} points!")

    await bot.process_commands(message)

# --- Run both web server and bot ---
async def main():
    await start_server()
    await bot.start(os.environ['TOKEN'])

asyncio.run(main())

File: bot.java
package com.mybot;

import net.dv8tion.jda.api.JDABuilder;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import net.dv8tion.jda.api.requests.GatewayIntent;

import com.sun.net.httpserver.HttpServer;
import java.net.InetSocketAddress;
import java.util.*;

public class Bot extends ListenerAdapter {

    private static Map<String, Integer> leaderboard = new HashMap<>();

    public static void main(String[] args) throws Exception {
        String token = System.getenv("TOKEN");
        if (token == null || token.isEmpty()) {
            System.out.println("TOKEN missing!");
            return;
        }

        // --- Tiny HTTP server for Render ---
        HttpServer server = HttpServer.create(new InetSocketAddress(8080), 0);
        server.createContext("/", exchange -> {
            String response = "Bot is running!";
            exchange.sendResponseHeaders(200, response.getBytes().length);
            exchange.getResponseBody().write(response.getBytes());
            exchange.close();
        });
        server.start();
        System.out.println("Web server started on port 8080");

        // --- Start Discord bot ---
        JDABuilder.createDefault(token)
                .enableIntents(GatewayIntent.MESSAGE_CONTENT, GatewayIntent.GUILD_MESSAGES)
                .addEventListeners(new Bot())
                .build()
                .awaitReady(); // Important for Render

        System.out.println("Discord bot is online!");
    }

    @Override
    public void onMessageReceived(MessageReceivedEvent event) {
        if (event.getAuthor().isBot()) return;

        String msg = event.getMessage().getContentRaw();
        String userId = event.getAuthor().getId();

        // --- Leaderboard logic ---
        leaderboard.put(userId, leaderboard.getOrDefault(userId, 0) + 1);

        if (msg.equalsIgnoreCase(".lb")) {
            StringBuilder sb = new StringBuilder("🏆 Leaderboard 🏆\n");
            leaderboard.entrySet().stream()
                    .sorted((a, b) -> b.getValue() - a.getValue())
                    .limit(10)
                    .forEach(entry -> sb.append("<@")
                            .append(entry.getKey())
                            .append("> - ")
                            .append(entry.getValue())
                            .append(" pts\n"));
            event.getChannel().sendMessage(sb.toString()).queue();
        }

        if (msg.equalsIgnoreCase(".points")) {
            int pts = leaderboard.getOrDefault(userId, 0);
            event.getChannel().sendMessage("You have " + pts + " points!").queue();
        }

        // Add other old commands here if needed...
    }
}
